exchanges/graphcache/src/types.ts

```diff
@@ -128,7 +128,7 @@ export type Resolver = (
   args: Variables,
   cache: Cache,
   info: ResolveInfo
-) => DataField | undefined;
+) => DataField | (DataFields & { __typename?: string }) | undefined;
```

Comment:
I believe we also forgot `null` here? Let's also make a nice alias for this like `ResolverResult`?